### ISSUE TITLE
Remove manual check for roles on document form

### ISSUE THREAD

**Issue by emmajane** [humanitarianresponse/site#264]
While updating a document as a Site Administrator, I have discovered that the Clusters/Sectors field has hard-coded roles (see snippet below). As a result, if you want to edit a Document in a group, you have to first add yourself to the group, assign yourself the role of manager, and then you can save the form. The alternative shortcut is to masquerade as an existing Group Manager to make the edit.

# Notes

Sample URL: https://www.humanitarianresponse.info/en/operations/whole-of-syria/document/2016-syrian-arab-republic-humanitarian-response-plan/edit

This hard-coded snippet results in "Your groups" being a required field:

    if (!in_array('manager', $roles) && !in_array('editor', $roles) && !in_array('contributor', $roles) && $op->uid != $user->uid && !in_array('editor', $user->roles)) {

If you are a Site Administrator, "Other groups" appears. This is not related to "Your groups" being a required field. It's a side effect of being Site Administrator.

![2015-12-30 at 15 22](https://cloud.githubusercontent.com/assets/1299320/12052864/355c5860-af0b-11e5-9527-d3459c66ab8e.png)

# Fix

A clean fix for this issue would be:

- Create a new permission for forcing "Your groups" to be a required field.
- Assign relevant roles to this permission (including site Administrator role)
- Remove the manual check for roles (`// If user is not a manager or editor of operation, make bundles field required`)

**Comment by pcambra** [comment]
Agreed, however, there might be more stuff behind this access logic, as the field is hidden also based on permissions and prepopulation conditions. We should fully understand what's intended with the permission check and show/hide for the field_bundles form before changing it.
